pbm/storage/s3/s3.go

```diff
@@ -617,3 +639,24 @@ func awsLogger(l log.LogEvent) aws.Logger {
 		l.Debug(msg, xs...)
 	})
 }
+
+func onGCE() bool {
+	client := http.Client{
+		Timeout: 100 * time.Millisecond,
+	}
+
+	req, err := http.NewRequest("GET", "http://169.254.169.254/computeMetadata/v1", nil)
```

Comment:
if we add the trailing slash, client doesn't need to follow redirect

```suggestion
	req, err := http.NewRequest("GET", "http://169.254.169.254/computeMetadata/v1/", nil)
```